src/util.rs

```diff
@@ -61,8 +61,25 @@ fn exit_with_code(r: reqwest::StatusCode) {
 
 // todo : assume https as the default scheme
 // Or get rid of this.
-pub fn url_validation(url: &str) -> Result<Url> {
-    Url::parse(url).context(format!("URL args: \'{}\' is not valid", url))
+pub fn url_validation(url: &str) -> Url {
+    let temp_url = Url::parse(url);
+    let temp_url = match temp_url {
+        Ok(pass) => {
+            https(pass)
+        },
+        Err(_error) => {
+            Url::parse(&format!("https://{}",url)).unwrap()
+        },
+    };
+    temp_url
+}
+
+fn https(url: Url) -> Url {
+    if url.as_str().starts_with("http://") {
+        Url::parse(&format!("https://{}",url.host_str().unwrap())).unwrap()
```

Comment:
You can't force https like this. If the user have inputed an URL with `http` there may be goods reasons (the deployment may not support https)

Reply:
got it

Reply:
Url validation work is to check whether url is correct or not. 
And If if schema is missing the force https into the url.